mlflow/tracing/processor/base_mlflow.py

```diff
@@ -108,17 +106,19 @@ def _get_experiment_id_for_trace(self, span: OTelReadableSpan) -> str:
 
         The experiment ID can be configured in multiple ways, in order of precedence:
           1. An experiment ID specified via the span creation API i.e. MlflowClient().start_trace()
-          2. An experiment ID specified via the processor constructor
+          2. An experiment ID specified via `mlflow.tracing.set_destination`
           3. An experiment ID of an active run.
           4. The default experiment ID
         """
+        from mlflow.tracing.provider import _MLFLOW_TRACE_USER_DESTINATION
         from mlflow.tracking.fluent import _get_latest_active_run
 
         if experiment_id := get_otel_attribute(span, SpanAttributeKey.EXPERIMENT_ID):
             return experiment_id
 
-        if self._experiment_id:
-            return self._experiment_id
+        if destination := _MLFLOW_TRACE_USER_DESTINATION.get():
+            if destination.experiment_id:
+                return destination.experiment_id
```

Comment:
```suggestion
            if exp_id := getattr(destination, "experiment_id"):
                return exp_id
```

nit: we may add a destination that doesn't add `experiment_id` field